tools/checklocks/state.go

```diff
@@ -32,9 +32,64 @@ type lockType int
 const (
 	unlocked lockType = iota
 	locked
-	// TODO(jamesyou): DEFERUNLOCK
+	deferredunlock
+	lockedAndDeferredUnlock
```

Comment:
Should there be an `error` state?

Reply:
Not unless you need one.

Reply:
Few points. Some analyses continue doing their work when they see an error. For such analyses, it makes sense to have error as state. For other analyses that just exit once they see an error, the error state is probably unnecessary.